airflow/www_rbac/decorators.py

```diff
@@ -86,3 +88,44 @@ def zipper(response):
         return f(*args, **kwargs)
 
     return view_func
+
+
+def has_dag_access(**dag_kwargs):
+    """
+    Decorator to check whether the user has read / write permission on the dag.
+    """
+    def decorator(f):
+        @functools.wraps(f)
+        def wrapper(self, *args, **kwargs):
+            dag_id = request.args.get('dag_id')
+            # if it is true, we need to check whether user has read access on the dag
+            can_dag_read = dag_kwargs.get('can_dag_read', False)
+            # if it is false, we need to check whether user has write access on the dag
+            can_dag_edit = dag_kwargs.get('can_dag_edit', False)
+            if dag_id is None or (not can_dag_edit and not can_dag_read):
+                # this request is not dag-level related or no read/write level access perm required
```

Comment:
You might want to raise here, assuming only DAG-specific views would be decorated